compiler/rustc_parse_format/src/lib.rs

```diff
@@ -750,6 +757,32 @@ impl<'a> Parser<'a> {
         }
         if found { Some(cur) } else { None }
     }
+
+    fn suggest_positional_arg_instead_of_captured_arg(&mut self, arg: Argument<'a>) {
+        if let Some(end) = self.consume_pos('.') {
+            let byte_pos = self.to_span_index(end);
+            let start = InnerOffset(byte_pos.0 + 1);
+            let field = self.argument(start);
+            if !self.consume('}') {
```

Comment:
Can we add a comment stating "We can only parse `foo.bar` field access, any deeper nesting, or another type of expression, like method calls, are not supported".